### DISCUSSION TITLE
hdt-java vs hdt-cpp feature parity matrix

### DISCUSSION THREAD

**Discussion by davidandreoletti** [rdfhdt/hdt-java#168] [Q&A] [answered]
At first glance, both projects support HDT read,write,search operation.

However, I would like to know:
- does the Java version offload work to the cpp version ?
- if not, do you have a feature matrix to compare both projects ?

**Comment by D063520** [comment, marked as answer]
Hi,

the hdt-java and hdt-cpp implementations are completely independent (i.e. not calling each other) but compatible (i.e. and HDT generated with the c++ version can be read by the java version and viceversa).

About the feature matrix, there are many small differences. Generally the c++ version is a bit more efficient in compressing. For querying there are more features in the java version. But there is a big overlap. If you can tell what you want to do we can tell you.

**Reply by davidandreoletti** [reply]
Got it. Thanks.
